### ISSUE TITLE
C# algorithms are completely outdated

### ISSUE THREAD

**Issue by mnns** [hanabi1224/Programming-Language-Benchmarks#270]
No use of Memory<T> or Span<T> / ReadOnlyMemory etc. 
Code is non optimized...
Thanks

**Comment by bpecsek** [comment]
You can fork the repository, optimize the codes and send a PR.
If your codes are according to the rules the site maintainer accepts them and your new codes will be bench-marked.

**Comment by asaf92** [comment]
The binary tress benchmark was using structs but a PR removed them intentionally.
Is there a policy saying that you have to use unoptimized code or something similar?

**Comment by hanabi1224** [comment]
> The binary tress benchmark was using structs

@asaf92 It was removed because the 'optimization' is unfair in terms of heap allocation, it's also been removed from the CLBG site for a long time

**Comment by hanabi1224** [comment]
> It's a legitimate way to avoid garbage collection when writing high performance code.

Yes, except that this particular problem was originally designed to benchmark perf of boehm gc and heap allocations, the struct trick halves the number of heap allocations and just makes the comparison unfair and invalid

**Comment by JoaoVictorVP** [comment]
And what reason do you have for bringing it up? It's not like other languages don't have tricks like auto-allocating on the stack via escape parsing or compiler optimizations. Structs are extremely normal in C# and they are part of the code written everyday by millions of developers, I don't see how this is "unfair".
